### ISSUE TITLE
Request: function for getting "documents" dir

### ISSUE THREAD

**Issue by JPLeBreton** [ActiveState/appdirs#57]
eg c:\Users\user\Documents in Windows 7+, /Users/user/Documents in OSX, and ~/Documents in Linux, where applicable (XDG spec doesn't cover this).  The "winshell" module does this but only for Windows: https://github.com/tjguk/winshell

**Comment by JPLeBreton** [comment]
Link to a very easy, no-extra-modules-needed way to do this on Windows:
http://stackoverflow.com/a/30924555/1191587

In OSX and Linux this might be as simple as os.path.expanduser('~/Documents').

**Comment by jwilk** [comment]
For Linux, there's this [xdg-user-dirs](https://wiki.freedesktop.org/www/Software/xdg-user-dirs/) thing.
Unfortunately, I couldn't find any official comprehensive documentation for it. The best I could find is [an article on Arch Linux wiki](https://wiki.archlinux.org/index.php/Xdg_user_directories).

**Comment by JPLeBreton** [comment]
Like I said, XDG is about ~/.config, ~/.local etc and doesn't cover anything like a "Documents" directory, and there does not appear to be a formal standard for it in Linux, just a convention of ~/Documents used by many distros / DEs.  There is a standard for it in Windows and OSX though, and all together that seems solid enough to implement.

**Comment by jwilk** [comment]
For the avoidance of doubt, XDG Base Directory and xdg-user-dirs are not the same thing.
The former is about ~/.config and so on; the latter is about ~/Documents and friends.

**Comment by JPLeBreton** [comment]
Ah, I'm sorry I didn't read the Arch wiki page closely.  In that case it looks like checking for $XDG_DOCUMENTS_DIR would work?

**Comment by jwilk** [comment]
At least the [GLib implementation](https://git.gnome.org/browse/glib/tree/glib/gutils.c?id=2.45.8#n1503) doesn't inspect the environment; instead, it parses the `user-dirs.dirs` file directly.

**Comment by dgentry** [comment]
(After`import os`) It definitely works on Windows, Mac, and Ubuntu to use`os.path.expanduser("~/Documents")`.  On Mac OS X 12 (Sierra),
```
>>> os.path.expanduser("~/Documents")
'/Users/gentry/Documents'
```
On Windows 8.1,
```
>>> os.path.expanduser("~/Documents")
'C:\\Users\\IEUser/Documents'
>>> os.path.isdir(os.path.expanduser("~/Documents"))
True
```
On Ubuntu 16 LTS (xenial),
```
>>> os.path.expanduser("~/Documents")
'/home/vagrant/Documents'
```

**Comment by dgentry** [comment]
Also, I think this SHOULD be a function in appdirs.  In fact, I think it should be user_data_dir, and what is currently called user_data_dir should be called app_data_dir because the ~/Library/Application_Support/... (for example) is where the App stores data (hidden from the user by default). The USER stores data in their ~/Documents directory.

**Comment by Kriskras99** [comment]
I implemented this in my [fork](https://github.com/kriskras99/appdirs) and used the naming convention of xdg-users-dirs. I don't think user_data_dir should be renamed to app_data_dir because it would break backwards-compatibility and the naming does make sense, site_* is for data that should be accessible by all instances of the program no matter which user started it, while user_* is for user-specific data.

**Comment by dgentry** [comment]
Yeah, I understand that the original meaning was "site_" vs. "user_" and I
agree that we don't want to break backwards compatibility.  Since all three
OSes call it "Documents" maybe it should have "document" in its name.
Also, Windows and Mac people won't know what "xdg-" is.  "user-doc-dir?"

Thanks for the quick response!

**Comment by Kriskras99** [comment]
I do not fully understand your point.
xdg-users-dirs is just a standard for unix on which I based the function names.
You should look at my [appdirs.py](https://github.com/kriskras99/appdirs/blob/stable/appdirs.py#L421) from line 421 to 607.

**Comment by dgentry** [comment]
Thanks for the pointer to the code -- user_desktop_dir and
user_documents_dir are great names.  I'll go try them out in my app now.  :)

**Comment by LexiconCode** [comment]
I also believe this should be included. Microsoft defines `Per user “Documents”` “Document” type files that users create/open/close/save in the application that are per user.



Example: | MyDoc.doc
-- | --
Windows 7: | Libraries
Vista: | %USERPROFILE%\Document
XP: | %USERPROFILE%\My Documents
Environment Variable: | Not applicable
Known Folder ID: | FOLDERID_Documents
System.Environment.SpecialFolder: | System.Environment.SpecialFolder.MyDocuments
CSIDL: | CSIDL_MYDOCUMENTS, CSIDL_PERSONAL

[sources](https://blogs.msdn.microsoft.com/patricka/2010/03/18/where-should-i-store-my-data-and-configuration-files-if-i-target-multiple-os-versions/)
